src/smartcontracts/typesystem/typeExpressionParser.ts

```diff
@@ -26,6 +26,6 @@ export class TypeExpressionParser {
 
     private typeFormulaToType(typeFormula: TypeFormula): Type {
         const typeParameters = typeFormula.typeParameters.map((typeFormula) => this.typeFormulaToType(typeFormula));
-        return new Type(typeFormula.name, typeParameters, undefined, typeFormula.metadata);
+        return new Type(typeFormula.name, typeParameters, TypeCardinality.fixed(1), typeFormula.metadata);
```

Comment:
I was wrong before. Maybe `undefined` is better here (if all tests still pass). Cardinality == 1 is tricky (it's a white lie), since here we also create & return variadic types.

Can be either `undefined` of `fixed(1)` - since the latter is the default anyway. When applying the Python-like simplifications (in the not very close future), this will get simplified (no cardinality abstraction etc.).